pandas/core/arrays/boolean.py

```diff
@@ -131,11 +131,27 @@ def coerce_to_array(values, mask=None, copy=False):
         if copy:
             values = values.copy()
     else:
+        # TODO conversion from integer/float ndarray can be done more efficiently
+        #  (avoid roundtrip through object)
         values_object = np.asarray(values, dtype=object)
-        mask_values = isna(values)
+
+        inferred_dtype = lib.infer_dtype(values_object, skipna=True)
+        integer_like = ("floating", "integer", "mixed-integer-float")
+        if inferred_dtype not in ("boolean", "empty") + integer_like:
+            raise TypeError("Need to pass bool-like values")
+
+        mask_values = isna(values_object)
         values = np.zeros(len(values), dtype=bool)
         values[~mask_values] = values_object[~mask_values].astype(bool)
 
+        # if the values were integer-like, validate it were actually 0/1's
```

Comment:
Does this validate that they're actually 0 or 1? Don't you also need a check that the min / max is in that bound?

Reply:
Ah, sorry you've astyped to bool, so going back to float will make these 0 and 1.